### PR TITLE
Update vehicle schema

### PR DESCRIPTION
**Pull request by rymarczy**
Updated `config_rt_vehicle` to contain complete flat schema for gzip import files. 

Updated `record_from_entity` method to handle expected keys missing from import file. 

Updated file conversion tests to matched updated vehicle schema.

### CODE REVIEW

File: py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py [hunk 1]
```diff
@@ -11,47 +11,95 @@ def config_type(self) -> ConfigType:
...
+                ('schedule_relationship', pyarrow.string()),
+                ('start_date', pyarrow.string()),
+                ('start_time', pyarrow.string()),
+                ('trip_id', pyarrow.string()), # actual label: id
+                # entity -> vehicle -> vehicle
+                ('vehicle_id', pyarrow.string()), # actual label: id
+                ('vehicle_label', pyarrow.string()), # actual label: label
+                ('vehicle_consist', pyarrow.list_(
+                                        pyarrow.struct(
+                                            [pyarrow.field('label',pyarrow.string()),]
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py:46]: This schema should include all possible fields found in RT_Vehicle gzip files.

File: py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py [hunk 2]
```diff
@@ -11,47 +11,95 @@ def config_type(self) -> ConfigType:
...
+                ('start_date',),
+                ('start_time',),
+                ('id','trip_id'),
+            ),
+            'entity,vehicle,vehicle': (
+                ('id','vehicle_id',),
+                ('label','vehicle_label',),
+                ('consist','vehicle_consist',),
+            )
         }
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py:83]: Dictionary to handle transformation of original entity object to exported parquet format.

File: py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py [hunk 3]
```diff
@@ -11,47 +11,95 @@ def config_type(self) -> ConfigType:
...
+        def drill_entity(f):
+            ret_dict = entity
+            f_list = f.split(',')
+            if len(f_list) == 1:
+                return ret_dict
+            for k in f_list[1:]:
+                if ret_dict.get(k) is None:
+                    return None
+                ret_dict = ret_dict.get(k)
+            return ret_dict
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py:94]: Little function to parse dictionary transform object, not sure if this is the best approach, but we'll need something to handle expected entity keys not being present.

File: py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py [hunk 4]
```diff
@@ -11,47 +11,95 @@ def config_type(self) -> ConfigType:
...
+
+        record = {}
+        for drill_keys in transform_schema.keys():
+            pull_dict = drill_entity(drill_keys)
+            for get_field in transform_schema[drill_keys]:
+                if pull_dict is None:
+                    record[get_field[-1]] = None
+                else:
+                    record[get_field[-1]] = pull_dict.get(get_field[0])
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py:103]: Build return record with Dictionary transform object and `drill_entity` function.

File: py_gtfs_rt_ingestion/tests/test_ingest.py [hunk 5]
```diff
@@ -35,24 +35,26 @@ def test_file_conversion(tmpdir):
...
+        "schedule_relationship",
+        "start_date",
+        "start_time",
+        "trip_id",
         "vehicle_id",
         "vehicle_label",
-        'vehicle_timestamp',
-        # 'vehicle_consist_labels.list.item',
-        "consist_labels.list.item",
+        "vehicle_consist.list.item.label",
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/tests/test_ingest.py:57]: Update `paths_from_json` set to match updated RT_Vehicle export schema.

**Comment by mzappitello** [comment]
looks good to me.

pls do a squash and merge.

**Pull request merged**